extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -322,7 +322,7 @@ export abstract class AzureAuth implements vscode.Disposable {
 	 * @param azureResource
 	 * @returns The authentication result, including the access token
 	 */
-	public async getTokenMsal(accountId: string, azureResource: azdata.AzureResource, tenantId: string): Promise<AuthenticationResult | null> {
+	public async getTokenMsal(accountId: string, azureResource: azdata.AzureResource, tenantId: string): Promise<AuthenticationResult | azdata.ProviderError | null> {
```

Comment:
This should still be returning PromptFailedResult since it has the canceled property.